site/docs/hive.md

```diff
@@ -17,113 +17,211 @@
+
+Suppose there is a table `table_a` and the table location is `hdfs://some_path/table_a`. 
+Now overlay a Hive table on top of this Iceberg table by issuing Hive DDL like so:
+
 ```sql
-SELECT * from table_b;
+CREATE EXTERNAL TABLE table_a 
+STORED BY 'org.apache.iceberg.mr.hive.HiveIcebergStorageHandler' 
+LOCATION 'hdfs://some_bucket/some_path/table_a';
 ```
 
-#### Using Hadoop Catalog
-Iceberg tables created using `HadoopCatalog` are stored entirely in a directory in a filesystem like HDFS.
+#### Hadoop catalog tables
+
+Iceberg tables created using `HadoopCatalog` are stored entirely in a directory in a filesystem like HDFS, 
+similar to the tables created through `HadoopTables`.
+You can use other compute engines or Java/Python API to create such as table. 
 
-##### Create an Iceberg table
-The first step is to create an Iceberg table using the Spark/Java/Python API and `HadoopCatalog`. For the purposes of this documentation we will assume that the fully qualified table identifier is `database_a.table_c` and that the Hadoop Catalog warehouse location is `hdfs://some_bucket/path_to_hadoop_warehouse`. Iceberg will therefore create the table at the location `hdfs://some_bucket/path_to_hadoop_warehouse/database_a/table_c`.
+Suppose there is a table `table_b` and the table location is `hdfs://some_path/table_b`. 
 
-##### Create a Hive table
 Now overlay a Hive table on top of this Iceberg table by issuing Hive DDL like so:
+
 ```sql
-CREATE EXTERNAL TABLE database_a.table_c 
+CREATE EXTERNAL TABLE database_a.table_b
 STORED BY 'org.apache.iceberg.mr.hive.HiveIcebergStorageHandler' 
-LOCATION 'hdfs://some_bucket/path_to_hadoop_warehouse/database_a/table_c'
+LOCATION 'hdfs://some_path/table_b'
 TBLPROPERTIES (
   'iceberg.mr.catalog'='hadoop', 
-  'iceberg.mr.catalog.hadoop.warehouse.location'='hdfs://some_bucket/path_to_hadoop_warehouse')
-;
+  'iceberg.mr.catalog.hadoop.warehouse.location'='hdfs://some_bucket/path_to_hadoop_warehouse'
+);
 ```
+
 Note that the Hive database and table name *must* match the values used in the Iceberg `TableIdentifier` when the table was created. 
 
-It is possible to omit either or both of the table properties but instead you will then need to set these when reading from the table. Generally it is recommended to set them at table creation time so you can query tables created by different catalogs. 
+It is possible to omit either or both of the table properties but instead you will then need to set these when reading from the table.
+Generally it is recommended to set them at table creation time, so you can query tables created by different catalogs. 
+
+#### Hive catalog tables
+
+As described before, tables created by the `HiveCatalog` with Hive engine feature enabled are directly visible by the Hive engine, so there is no need to create an overlay.
+
+#### Custom catalog tables
+
+For a registered catalog, simply specify the catalog name in the statement using table property `iceberg.catalog`.
```

Comment:
Maybe we should mention, when the `iceberg.catalog` is not defined on table level and also the `iceberg.mr.catalog` is missing from the HadoopConfig, HiveCatalog will be used as default.